### PR TITLE
issue77: Fixed the invalid range error issue with the small matchee raw topic when compared with base raw topic length

### PR DESCRIPTION
**Pull request by ashu-pietech**
Added another condition to check if matchee topic's fragments length is less than the length of base topic's fragment. Currently there's only one condition to handle the opposite condition such as if base topic's fragment length is less than matchee fragment length

### CODE REVIEW

File: lib/src/mqtt_subscription_topic.dart [hunk 1]
```diff
@@ -95,6 +95,13 @@ class MqttSubscriptionTopic extends MqttTopic {
           return false;
         }
       }
+      // If we're at the last fragment of the matchee rawTopic but there are
+      // more fragments in the in the lhs rawTopic then the matchee rawTopic
```
Comment (shamblett) [inline lib/src/mqtt_subscription_topic.dart:99]: 'in the in the' typo

**Comment by shamblett** [comment]
Ok, the file mqtt_client_base_test.dart contains a group test ''Subscription Topic' to unit test this class, could you add a test/s to test your change please.

File: lib/src/mqtt_subscription_topic.dart (hunk 1 above)
Comment (ashu-pietech) [reply lib/src/mqtt_subscription_topic.dart:99]: Okay I'll fix it

**Comment by ashu-pietech** [comment]
> Ok, the file mqtt_client_base_test.dart contains a group test ''Subscription Topic' to unit test this class, could you add a test/s to test your change please.

Sure I'll check that

**Comment by shamblett** [comment]
Re released the package for issue 79, you will probably need to rebase.

**Comment by ashu-pietech** [comment]
@shamblett I've updated the PR after rebase

**Comment by shamblett** [comment]
OK an existing unit test is now failing, could you have a look please, thanks.

**Comment by ashu-pietech** [comment]
> OK an existing unit test is now failing, could you have a look please, thanks.

Okay I'll check it

**Comment by ashu-pietech** [comment]
@shamblett PR updated and now all the test cases are working

**Pull request merged**